Content/LeagueSandbox-Scripts/Buffs/Annie/EBuff.cs

```diff
@@ -0,0 +1,53 @@
+using GameServerCore.Enums;
+using GameServerCore.Domain.GameObjects;
+using GameServerCore.Domain.GameObjects.Spell;
+using LeagueSandbox.GameServer.GameObjects.Stats;
+using LeagueSandbox.GameServer.API;
+using static LeagueSandbox.GameServer.API.ApiFunctionManager;
+using GameServerCore.Scripting.CSharp;
+using GameServerCore.Domain;
+
+namespace Buffs
+{
+    internal class MoltenShield : IBuffGameScript
+    {
+        public BuffType BuffType => BuffType.COMBAT_ENCHANCER;
+        public BuffAddType BuffAddType => BuffAddType.RENEW_EXISTING;
+        public int MaxStacks => 1;
+        public bool IsHidden => false;
+
+        public IStatsModifier StatsModifier { get; private set; } = new StatsModifier();
+
+        public void OnActivate(IAttackableUnit unit, IBuff buff, ISpell ownerSpell)
+        {
+            float bonus = 10f + (10f * ownerSpell.CastInfo.Owner.Stats.Level);
+            
+            StatsModifier.Armor.FlatBonus += bonus;
+            StatsModifier.MagicResist.FlatBonus += bonus;
+            unit.AddStatModifier(StatsModifier);
+
+            ApiEventManager.OnBeingHit.AddListener(this, unit, OnBeingHit, false);
+        }
+
+        public void OnBeingHit(IAttackableUnit owner, IAttackableUnit attacker)
```

Comment:
Annie E utilizes OnPreDamage rather than OnBeingHit. It isn't necessary to change this now, but make a comment for later.
```suggestion
       // TODO: Use OnPreDamage instead.
        public void OnBeingHit(IAttackableUnit owner, IAttackableUnit attacker)
```